mos_tests/neutron/python_tests/test_dvr.py

```diff
@@ -125,37 +155,95 @@ def test_connectivity_after_reset_compute(self, env_name, variables):
         # reset compute
-        compute_hostname = getattr(server, 'OS-EXT-SRV-ATTR:host')
+        compute_hostname = getattr(self.server, 'OS-EXT-SRV-ATTR:host')
         self.reset_computes([compute_hostname], env_name)
 
         time.sleep(60)
 
-        self.check_ping_from_vm(server, vm_keypair=self.instance_keypair)
+        self.check_ping_from_vm(self.server, vm_keypair=self.instance_keypair)
+
+    def test_north_south_floating_ip_shut_down_br_ex_on_controllers(self):
+        """Check North-South connectivity with floatingIP after shut-downing
+        br-ex on all controllers
+
+        Scenario:
+            1. Create net01, subnet net01__subnet for it
+            2. Create router01 with external network and
+                router type Distributed
+            3. Add interfaces to the router01 with net01__subnet
+            4. Boot vm_1 in the net01
+            5. Associate floating IP
+            6. Go to the vm_1 with ssh and floating IP
+            7. Shut down br-ex on all controllers
+            8. Go to the vm_1 with ssh and floating IP
+            9. Ping 8.8.8.8
+
+        Duration 10m
+
+        """
+        self._prepare_openstack_env()
+
+        ip = self.floating_ip['floating_ip_address']
+        self.check_ping_from_vm_with_ip(ip, vm_keypair=self.instance_keypair,
+                                        ip_to_ping='8.8.8.8',
+                                        ping_count=10, vm_login='cirros')
+
+        self.shut_down_br_ex_on_controllers()
+
+        self.check_ping_from_vm_with_ip(ip, vm_keypair=self.instance_keypair,
+                                        ip_to_ping='8.8.8.8',
+                                        ping_count=10, vm_login='cirros')
+
+    def test_north_south_floating_ip_ban_clear_l3_agent_on_compute(self):
+        """Check North-South connectivity with floatingIP after ban and
+        clear l3-agent on compute
+
+        Scenario:
+            1. Create net01, subnet net01__subnet for it
+            2. Create router01 with external network and
+                router type Distributed
+            3. Add interfaces to the router01 with net01__subnet
+            4. Boot vm_1 in the net01
+            5. Associate floating IP
+            6. Go to the vm_1 with ssh and floating IP
+            7. Ping 8.8.8.8
+            8. Ban l3-agent on the compute with vm_1: service l3-agent stop
+            9. Wait 15 seconds
+            10. Clear this l3-agent: service l3-agent stop
+            11. Go to vm_1 with ssh and floating IP
+            12. Ping 8.8.8.8
+
+        Duration 10m
+
+        """
+        self._prepare_openstack_env()
+
+        ip = self.floating_ip['floating_ip_address']
+
+        self.check_ping_from_vm_with_ip(ip, vm_keypair=self.instance_keypair,
+                                        ip_to_ping='8.8.8.8',
+                                        ping_count=10, vm_login='cirros')
+
+        compute = self.env.find_node_by_fqdn(self.zone.hosts.keys()[0])
```

Comment:
You instance was booted on any compute (not only first). 
Use something like hostname from `self.server` to retrive compute fqdn